pkg/deployments/hardhat.config.ts

```diff
@@ -113,63 +113,71 @@ task('check-artifacts', `check that contract artifacts correspond to their build
-      if (args.id && args.name) {
-        const task = new Task(args.id, TaskMode.READ_ONLY, hre.network.name);
-        await saveActionIds(task, args.name);
-        return;
-      }
+        async function generateActionIdsForTask(taskId: string): Promise<void> {
+          const task = new Task(taskId, TaskMode.READ_ONLY, hre.network.name);
+          const outputDir = path.resolve(task.dir(), 'output');
 
-      async function generateActionIdsForTask(taskId: string): Promise<void> {
-        const task = new Task(taskId, TaskMode.READ_ONLY, hre.network.name);
-        const outputDir = path.resolve(task.dir(), 'output');
+          if (existsSync(outputDir) && statSync(outputDir).isDirectory()) {
+            for (const outputFile of readdirSync(outputDir)) {
+              const outputFilePath = path.resolve(outputDir, outputFile);
+              if (outputFile.includes(hre.network.name) && statSync(outputFilePath).isFile()) {
+                const fileContents = JSON.parse(readFileSync(outputFilePath).toString());
+                const contractNames = Object.keys(fileContents).filter((name) => name !== 'timestamp');
 
-        if (existsSync(outputDir) && statSync(outputDir).isDirectory()) {
-          for (const outputFile of readdirSync(outputDir)) {
-            const outputFilePath = path.resolve(outputDir, outputFile);
-            if (outputFile.includes(hre.network.name) && statSync(outputFilePath).isFile()) {
-              const fileContents = JSON.parse(readFileSync(outputFilePath).toString());
-              const contractNames = Object.keys(fileContents).filter((name) => name !== 'timestamp');
-
-              for (const contractName of contractNames) {
-                await saveActionIds(task, contractName);
+                for (const contractName of contractNames) {
+                  await saveActionIds(task, contractName);
+                }
               }
             }
           }
         }
-      }
 
-      if (args.id) {
-        await generateActionIdsForTask(args.id);
-        return;
-      }
+        if (args.id) {
+          await generateActionIdsForTask(args.id);
+          return;
+        }
 
-      // We're calculating action IDs for whichever contracts we can pull enough information from disk for.
-      // This will calculate action IDs for any contracts which are a named output from a task.
-      for (const taskID of Task.getAllTaskIds()) {
-        await generateActionIdsForTask(taskID);
+        // We're calculating action IDs for whichever contracts we can pull enough information from disk for.
+        // This will calculate action IDs for any contracts which are a named output from a task.
+        for (const taskID of Task.getAllTaskIds()) {
+          await generateActionIdsForTask(taskID);
+        }
       }
+
+      await saveActionIdsTask(args, hre);
+      checkActionIdUniqueness(hre.network.name);
```

Comment:
Is this because of the early returns?

Reply:
yes, it is.

I considered creating subtasks, but it seemed an overkill. I think this is a better way to put it: first do the original task (whatever the execution path is), then check uniqueness.